Core/clim-basic/windowing/events.lisp

```diff
@@ -69,6 +69,7 @@
    ;; This slot is pretty much required in order to call handle-event. Some
    ;; events have something other than a sheet in this slot, which is gross.
    (sheet :initarg :sheet
+          :initform nil
           :reader event-sheet)))
```

Comment:
Can we add `(:default-initargs :sheet (alexandria:required-argument :sheet))` sheet here and do `:sheet nil` as a workaround in EAS?

Reply:
If possible as a separate commit since both changes are rather severe and separating them is better for bisecting, CI, reviewing, etc.

Reply:
this will break i.e throw-object-ptype, so I'd rather keep it as it is now (so we change a default behavior as little as possible) -- this is mostly to address the issue of incorrect coordinates for scrolling, not to fix the "gross" behavior of putting something what is not a sheet in the event.

Reply:
If the sheet is required by the spec (and code unprepared to handle `event-sheet` returning `nil`) I don't think we should further accommodate not supplying the `:sheet` initarg. That is, I would prefer not to add this default or the conditional in the `initialize-instance` method for `device-event`.

If there are things besides the one instance in ESA that would definitely break, things may be different, but do we know of any?

Reply:
checking for nil sheet is still necessary in initialize-instance, because some functions simply pass supplied sheet argument, i.e throw-ptype (not to mention ESA). I've made the initarg required and kept the test.